src/components/CharacterForm/CharacterForm.js

```diff
@@ -81,25 +104,43 @@ const CharacterForm = ({ onClose }) => {
     numberOfFilms,
   }) => {
     dispatch(
-      addCharacter({
-        id: nanoid(),
-        name,
-        eyeColor,
-        height,
-        gender: gender?.value,
-        birthYear,
-        homeworld: {
-          id: nanoid(),
-          name: homeworld,
-        },
-        species: {
-          id: nanoid(),
-          name: species?.value,
-        },
-        filmConnection: {
-          totalCount: numberOfFilms,
-        },
-      })
+      isEdit
+        ? editCharacter({
+            id: character.id,
+            name,
+            eyeColor,
```

Comment:
This function is a lot larger than it needs to be because you are repeating yourself, how can you simplify this and remove repetition?